pep-0622.rst

```diff
@@ -22,88 +22,139 @@ Resolution:
 Abstract
 ========
 
-This PEP proposes to add a pattern matching statement to Python,
-inspired by similar syntax found in Scala and many other languages.
-
-The pattern syntax builds on Python’s existing syntax for sequence
-unpacking (e.g., ``a, b = value``), but is wrapped in a ``match``
-statement which compares its subject to several different “shapes”
-until one is found that fits. In addition to specifying the shape of a
-sequence to be unpacked, patterns can also specify the shape to be a
-mapping with specific keys, an instance of a given class with (optionally) specific
-attributes, a specific value, or a wildcard. Patterns can be composed
-in several ways.
-
-Syntactically, a ``match`` statement contains a *subject* expression
-and one or more ``case`` clauses, where each case clause specifies a
-pattern (the overall shape to be matched), an optional “guard” (a
-condition to be checked if the pattern matches), and a code block to
-be executed if the case clause is selected.
-
-The rest of the PEP motivates why we believe pattern matching makes a
-good addition to Python, explains our design choices, and contains a
-precise syntactic and runtime specification. We also give guidance for
-static type checkers (and one small addition to the ``typing`` module)
-and discuss the main objections and alternatives that have been
-brought up during extensive discussion of the proposal, both within
-the group of authors and in the python-dev community. Finally, we
-discuss some possible extensions that might be considered in the
-future, once the community has ample experience with the currently
-proposed syntax and semantics.
+This PEP proposes to add a **pattern matching statement** to Python,
+inspired by similar syntax found in Scala, Erlang, and other languages.
 
+Patterns and shapes
+-------------------
+
+The **pattern syntax** builds on Python’s existing syntax for sequence
+unpacking (e.g., ``a, b = value``).
```

Comment:
```suggestion
unpacking (e.g., ``a, b = subject``).
```
This could make it easier to understand what we mean by "subject" below

Reply:
I'm not so sure; at this point the user should know what "value" stands for in the example, is but they haven't been introduced to "subject". The following paragraph in effect *introduces* the concept of "subject" (implicitly by naming it in context). If you think it requires more introduction we should somehow emphasize it in the next paragraph, e.g. "A ``match`` statement compares a value (the **subject**) to several different ..." (Note that I also don't like the repetition of **match** in bold.)

Reply:
Whoops, I accidentally committed your suggestion. I've reverted that commit.

Reply:
I was assuming here that it helps introduce subject because the reader could deduce from its previously known context (i.e. sequence unpacking). But again, I'll leave it to a subjective (pun not intended) taste.

Regarding the double bolding of **match**, do you mean **shape**? I would unbold at least one, I tend to use bold when introducing a new term (helps look up when rereading), but it shouldn't be needed the second time.